lib/mailbox-handler.js

```diff
@@ -201,10 +222,16 @@ class MailboxHandler {
                     return callback(err);
                 }
                 if (!mailboxData) {
-                    return callback(null, 'NONEXISTENT');
+                    const err = new Error('Mailbox deletion failed with code NoSuchMailbox');
+                    err.code = 'NONEXISTENT';
+                    err.responseCode = 404;
+                    return callback(err, 'NONEXISTENT');
                 }
                 if (mailboxData.specialUse || mailboxData.path === 'INBOX' || mailboxData.hidden) {
-                    return callback(null, 'CANNOT');
+                    const err = new Error('Mailbox deletion failed with code DisallowedMailboxMethod');
+                    err.code = 'CANNOT';
+                    err.responseCode = 405;
```

Comment:
405 is not a good choice, even if the status response message would seem like it. 405 covers HTTP methods, not API methods, i.e. you should get it when running PUT to a path that only allows GET requests. There are also [additional requirements](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405) for using this response code (has to return a `Allow` header)

Use a generic error code like 400 instead for the 405 responses.

Reply:
Fixed. Made 405 errors to be 400 errors, fixed tests accordingly.